src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+abstract class Figure implements AreaCalculator, DrawInfo {
+    private String color;
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < figures.length / 2; i++) {
+            figures[i] = figureSupplier.getRandomFigure(); // generation random figures for array
+        }
+
+        // generation default figure for array
+        for (int i = figures.length / 2; i < figures.length; i++) {
+            figures[i] = figureSupplier.getDefaultFigure();
+        }
+
+        for (Figure figure : figures) {
+            figure.toDraw();
+        }
+    }
```

Comment:
Don't extend your `Main/Application` class from `FigureSupplier` or `ColorSupplier`. The `main` method logic should be placed in a separate class, not within the `Figure` class.